src/lib.rs

```diff
@@ -40,10 +44,46 @@
 #[cfg(all(test, feature = "unstable"))] extern crate test;
 
 extern crate arrayvec;
+extern crate sr_std as rstd;
 
+#[macro_use]
+extern crate cfg_if;
+cfg_if! {
+    if #[cfg(any(all(target_arch = "wasm32", not(target_os = "emscripten"))))] {
+        mod libc {
+            #![allow(non_camel_case_types)]
+
+            // Use repr(u8) as LLVM expects `void*` to be the same as `i8*` to help enable
+            // more optimization opportunities around it recognizing things like
+            // malloc/free.
+            #[repr(u8)]
+            pub enum c_void {
```

Comment:
Hmm even `libc` does not define `c_void` for `wasm`. Do we really need it? Can we not remove it somehow?